test/test_linear_algebra.jl

```diff
@@ -94,4 +94,44 @@ fs =  parsed_args["fs"]
         @test isapprox(norm(w_out.weights[1][:, 1:19]), 0f0; rtol=ftol)
         mul!(w_out, M', w)
         @test isapprox(w_out, M'*w; rtol=ftol)
+
+        # test in-place
+        
+        dobs1 = deepcopy(dobs)
+        for Op in [F, F', Md , Md']
+                m = Op*dobs 
+                # Test that dobs wasn't modified
+                @test isapprox(dobs,dobs1,rtol=eps())
+                # Test that it did compute something 
+                @test m != dobs
+        end
+
+        w1 = deepcopy(w)
+        for Op in [D, D', M , M']
+                m = Op*w
+                # Test that w wasn't modified
+                @test isapprox(w,w1,rtol=eps())
+                # Test that it did compute something 
+                @test m != w
+        end
+
+        
+        n = (10,10,10)
+        d = (10.,10.,10.)
+        o = (0.,0.,0.)
+        m = 0.25*ones(Float32,n)
+        model3D = Model(n,d,o,m)
+
+        D3 = judiDepthScaling(model3D)
+
+        w2 = randn(Float32,prod(n))
+        w3 = deepcopy(w2)
+        for Op in [D3, D3']
+                m = Op*w2
+                # Test that w wasn't modified
+                @test w3 == w2
+                # Test that it did compute something 
+                @test m != w2
```

Comment:
test value of result as well

Reply:
added value test of depth scaling and top mute